### PR TITLE
[2.0.0] [Berlin/UK] Kubernetes Artifacts for WSO2 Open Banking Pattern 2

### PR DESCRIPTION
**Pull request by nalakagunathilaka**
## Purpose
This PR performs $subject adding the necessary Kubernetes and Helm resources for the deployment. This fixes following issues #68 #69 #70 #71.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,26 +1,24 @@
 # Changelog
 
-All notable changes to Kubernetes and Helm resources for WSO2 API Management version `1.5.x` in each resource release,
+All notable changes to Kubernetes and Helm resources for WSO2 Open Banking version `2.0.0` in each resource release,
```
Comment (Aaquiff) [inline CHANGELOG.md]: Should be 2.0.x, since the old one is 1.5.x.

File: database/mysql-ob/Chart.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+# Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Aaquiff) [inline database/mysql-ob/Chart.yaml]: No need to update the header since this is an existing file. The same applies to all files that are moved.

File: CHANGELOG.md (hunk 1 above)
Comment (nalakagunathilaka) [reply CHANGELOG.md]: Created branch is 2.0.0 and previous branches were created in the format of 1.4/5.x. That's the reason.

File: ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-synapse-configs-token-api.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+  {{ if index .Values "mysql-ob" "wso2" "deployment" "spec" "uk" "enabled" }}
+# Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Aaquiff) [inline ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-synapse-configs-token-api.yaml]: Update header

File: database/mysql-ob/Chart.yaml (hunk 2 above)
Comment (nalakagunathilaka) [reply database/mysql-ob/Chart.yaml]: Fixed with 0a68c76.

File: ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-synapse-configs-token-api.yaml (hunk 3 above)
Comment (nalakagunathilaka) [reply ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-synapse-configs-token-api.yaml]: Fixed with 0a68c76.

File: CHANGELOG.md (hunk 1 above)
Comment (chirangaalwis) [reply CHANGELOG.md]: Let's go ahead with `2.0.x` because we will eventually create this branch as the development branch. `2.0.0` is only a temporary branch.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,26 +1,24 @@
...
 will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
-## [v1.5.0.1] - 2020-09-25
+## [v2.0.0.1] - 2021-01-22
```
Comment (chirangaalwis) [inline CHANGELOG.md]: @nalakagunathilaka let's set the date just prior to sending the `master` merge PR.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -23,21 +23,21 @@ The issues page on GitHub is for tracking bugs and feature requests. When posing
 
 If you like to contribute with a bug fix or a new feature, start by posting an issue and discussing the best way to implement it.
 
-Unlike most projects, development for this repository is carried out on the `1.5.x` branch. This is because the `master` branch contains the latest stable release of the project.
-The code in `1.5.x` is merged to the `master` branch after a final review and a round of testing.
+Unlike most projects, development for this repository is carried out on the `2.0.0` branch. This is because the `master` branch contains the latest stable release of the project.
```
Comment (chirangaalwis) [inline CONTRIBUTING.md]: @nalakagunathilaka let's set the branch to `2.0.x`.

File: database/mysql-ob-berlin/requirements.yaml [hunk 6]
```diff
@@ -11,9 +11,7 @@
...
-
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: {{ .Values.kubernetes.serviceAccount }}
-  namespace : {{ .Release.Namespace }}
+dependencies:
+  - name: mysql
+    version: "1.6.9"
+    repository: "https://helm.wso2.com"
```
Comment (chirangaalwis) [inline database/mysql-ob-berlin/requirements.yaml:17]: @nalakagunathilaka @Aaquiff have we uploaded the base MySQL Helm chart to `https://helm.wso2.com` for this to work?

File: database/mysql-ob/Chart.yaml [hunk 7]
```diff
@@ -16,4 +16,4 @@ apiVersion: v1
 appVersion: "5.7"
 description: A Helm chart for MySQL based deployment of WSO2 Open Banking Datasources
 name: mysql-ob
-version: 1.5.0-1
+version: 2.0.0-1
```
Comment (chirangaalwis) [inline database/mysql-ob/Chart.yaml]: @nalakagunathilaka please add a new EOF line.

File: database/mysql-ob/requirements.yaml [hunk 8]
```diff
@@ -12,16 +12,12 @@
...
-    protocol: TCP
+dependencies:
+  - name: mysql-ob-berlin
+    version: "2.0.0-1"
+    repository: "file://../mysql-ob-berlin"
+    condition: wso2.deployment.spec.berlin.enabled
+  - name: mysql-ob-uk
+    version: "2.0.0-1"
+    repository: "file://../mysql-ob-uk"
+    condition: wso2.deployment.spec.uk.enabled
```
Comment (chirangaalwis) [inline database/mysql-ob/requirements.yaml]: @nalakagunathilaka let's add a new EOF line.

File: database/mysql-ob/requirements.yaml [hunk 9]
```diff
@@ -12,16 +12,12 @@
...
-    deployment: wso2ob-km
-  sessionAffinity: ClientIP
-  ports:
-  - name: servlet-https
-    port: 9446
-    protocol: TCP
+dependencies:
+  - name: mysql-ob-berlin
+    version: "2.0.0-1"
+    repository: "file://../mysql-ob-berlin"
```
Comment (chirangaalwis) [inline database/mysql-ob/requirements.yaml:18]: @nalakagunathilaka let's not commit the file path based repository value. Let's set the expected, final Helm chart name in the official WSO2 Helm chart repository for each sub chart in this case.

Also, let's point to the official WSO2 Helm chart repository.

Once, the charts have been officially uploaded, please perform a test to ensure whether the added dependencies are properly downloaded when building this chart.

File: database/mysql-ob/values.yaml [hunk 10]
```diff
@@ -11,7 +11,10 @@
...
-  - name: mysql
-    version: "1.6.7"
-    repository: "https://kubernetes-charts.storage.googleapis.com"
+wso2:
+  deployment:
+    spec:
+      berlin:
+        enabled: true
+      uk:
+        enabled: false
```
Comment (chirangaalwis) [inline database/mysql-ob/values.yaml]: @nalakagunathilaka please add a new EOF line.

File: ob-pattern-2/requirements.yaml [hunk 11]
```diff
@@ -14,8 +14,8 @@
 
 dependencies:
   - name: mysql-ob
-    version: "1.5.0-1"
-    repository: "https://helm.wso2.com"
+    version: "2.0.0-1"
+    repository: "file://../database/mysql-ob"
```
Comment (chirangaalwis) [inline ob-pattern-2/requirements.yaml:18]: @nalakagunathilaka let's revert this change as we expect the pull the chart from the official WSO2 Helm chart repository, once uploaded.

File: ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-jaggeryapp-devportal.yaml [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+            spec: 'UK',
+            grantTypes: {
+                authorization_code: 'Code',
+                implicit: 'Implicit',
+                refresh_token: 'Refresh Token',
+                client_credentials: 'Client Credentials',
+            },
+        },
+    };
+
```
Comment (chirangaalwis) [inline ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-jaggeryapp-devportal.yaml]: @nalakagunathilaka let'e remove these additional, unnecessary EOF lines.

File: ob-pattern-2/templates/ob-bi/dashboard/wso2ob-pattern-2-ob-bi-dashboard-bin.yaml [hunk 13]
```diff
@@ -298,21 +309,23 @@ data:
...
         -Djavax.net.ssl.keyStore="$CARBON_HOME/resources/security/wso2carbon.jks" \
         -Djavax.net.ssl.keyStorePassword="wso2carbon" \
         -Djavax.net.ssl.trustStore="$CARBON_HOME/resources/security/client-truststore.jks" \
         -Djavax.net.ssl.trustStorePassword="wso2carbon" \
+        -Dio.netty.tryReflectionSetAccessible=false \
         org.wso2.carbon.launcher.Main $*
         status=$?
     done
-    {{ end }}
+
```
Comment (chirangaalwis) [inline ob-pattern-2/templates/ob-bi/dashboard/wso2ob-pattern-2-ob-bi-dashboard-bin.yaml]: @nalakagunathilaka let's remove this additional EOF line. Please check other files for similar cases and remove them if any.

**Review by chirangaalwis — CHANGES_REQUESTED** [review]

File: database/mysql-ob-berlin/requirements.yaml (hunk 6 above)
Comment (nalakagunathilaka) [reply database/mysql-ob-berlin/requirements.yaml:17]: Yes and I tested it.

File: database/mysql-ob/requirements.yaml (hunk 9 above)
Comment (nalakagunathilaka) [reply database/mysql-ob/requirements.yaml:18]: It was decided to keep local repository paths like that. @Aaquiff please comment on this.

File: ob-pattern-2/requirements.yaml (hunk 11 above)
Comment (nalakagunathilaka) [reply ob-pattern-2/requirements.yaml:18]: It was decided to keep local repository paths like that and not push the chart to Helm Repo. @Aaquiff please comment on this.

File: CHANGELOG.md (hunk 4 above)
Comment (nalakagunathilaka) [reply CHANGELOG.md]: Will set a final commit to set the date when all comments are resolved and ready merge to send the merge PR to master as well.

File: ob-pattern-2/templates/ob-bi/dashboard/wso2ob-pattern-2-ob-bi-dashboard-bin.yaml (hunk 13 above)
Comment (nalakagunathilaka) [reply ob-pattern-2/templates/ob-bi/dashboard/wso2ob-pattern-2-ob-bi-dashboard-bin.yaml]: Fixed with bfff48b.

File: ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-jaggeryapp-devportal.yaml (hunk 12 above)
Comment (nalakagunathilaka) [reply ob-pattern-2/templates/ob-am/wso2ob-pattern-2-ob-am-conf-jaggeryapp-devportal.yaml]: Fixed with bfff48b.

File: database/mysql-ob/values.yaml (hunk 10 above)
Comment (nalakagunathilaka) [reply database/mysql-ob/values.yaml]: Fixed with bfff48b.

File: database/mysql-ob/requirements.yaml (hunk 8 above)
Comment (nalakagunathilaka) [reply database/mysql-ob/requirements.yaml]: Fixed with bfff48b.

File: database/mysql-ob/Chart.yaml (hunk 7 above)
Comment (nalakagunathilaka) [reply database/mysql-ob/Chart.yaml]: Fixed with bfff48b.

File: CONTRIBUTING.md (hunk 5 above)
Comment (nalakagunathilaka) [reply CONTRIBUTING.md]: Fixed with bfff48b.

File: CHANGELOG.md (hunk 1 above)
Comment (nalakagunathilaka) [reply CHANGELOG.md]: Fixed with bfff48b.

File: database/mysql-ob-berlin/requirements.yaml (hunk 6 above)
Comment (Aaquiff) [reply database/mysql-ob-berlin/requirements.yaml:17]: @chirangaalwis it has been uploaded.

File: database/mysql-ob/requirements.yaml (hunk 9 above)
Comment (Aaquiff) [reply database/mysql-ob/requirements.yaml:18]: @chirangaalwis since these charts are not used standalone, do we need to publish this as a separate chart?

File: CHANGELOG.md (hunk 4 above)
Comment (chirangaalwis) [reply CHANGELOG.md]: +1

File: database/mysql-ob/requirements.yaml (hunk 9 above)
Comment (chirangaalwis) [reply database/mysql-ob/requirements.yaml:18]: Yeah. +1, let's go with the file path in this chart.

File: ob-pattern-2/requirements.yaml (hunk 11 above)
Comment (chirangaalwis) [reply ob-pattern-2/requirements.yaml:18]: +1. @Aaquiff I suppose then we may have to follow the same approach in other product charts as well.

**Review by chirangaalwis — APPROVED** [review]
